core/services/vrf/listener_v2.go

```diff
@@ -321,141 +378,251 @@ func (lsn *listenerV2) processRequestsPerSub(
-		rlog := lggr.With(
-			"reqID", vrfRequest.RequestId.String(),
-			"txHash", vrfRequest.Raw.TxHash,
-			"maxGasPrice", maxGasPriceWei.String(),
-			"fromAddress", fromAddress)
 
 		// This check to see if the log was consumed needs to be in the same
 		// goroutine as the mark consumed to avoid processing duplicates.
 		consumed, err := lsn.logBroadcaster.WasAlreadyConsumed(req.lb)
 		if err != nil {
 			// Do not process for now, retry on next iteration.
-			rlog.Errorw("Could not determine if log was already consumed", "error", err)
-			continue
+			l.Errorw("Could not determine if log was already consumed",
+				"reqID", req.req.RequestId.String(),
+				"txHash", req.req.Raw.TxHash,
+				"error", err)
 		} else if consumed {
-			processed[vrfRequest.RequestId.String()] = struct{}{}
-			continue
+			processed[req.req.RequestId.String()] = struct{}{}
+		} else {
+			unconsumed = append(unconsumed, req)
 		}
+	}
 
-		// Check if we can ignore the request due to it's age.
-		if time.Now().UTC().Sub(req.utcTimestamp) >= lsn.job.VRFSpec.RequestTimeout {
-			rlog.Infow("Request too old, dropping it")
-			lsn.markLogAsConsumed(req.lb)
-			processed[vrfRequest.RequestId.String()] = struct{}{}
-			incDroppedReqs(lsn.job.Name.ValueOrZero(), lsn.job.ExternalJobID, v2, reasonAge)
-			continue
+	// Process requests in chunks
+	for chunkStart := 0; chunkStart < len(unconsumed); chunkStart += int(lsn.job.VRFSpec.ChunkSize) {
+		chunkEnd := chunkStart + int(lsn.job.VRFSpec.ChunkSize)
+		if chunkEnd > len(unconsumed) {
+			chunkEnd = len(unconsumed)
 		}
-
-		// Check if the vrf req has already been fulfilled.
-		// Pass max(request.BlockNumber, currentBlockNumber) to the call to ensure we get
-		// the latest state or an error.
-		// This is to avoid a race condition where a request commitment has been written on-chain
-		// but has not been synced fully on the entire network. GetCommitment will return an empty
-		// byte array in that case, and we would think that the request is fulfilled when it actually
-		// isn't.
-		reqBlockNumber := new(big.Int).SetUint64(vrfRequest.Raw.BlockNumber)
-		currBlock := new(big.Int).SetUint64(lsn.getLatestHead())
-		m := bigmath.Max(reqBlockNumber, currBlock)
-		callback, err := lsn.coordinator.GetCommitment(&bind.CallOpts{
-			BlockNumber: m,
-		}, vrfRequest.RequestId)
-		if err != nil {
-			rlog.Warnw("Unable to check if already fulfilled, processing anyways",
-				"err", err,
-				"reqBlockNumber", reqBlockNumber,
-				"currBlock", currBlock,
-			)
-		} else if utils.IsEmpty(callback[:]) {
-			// If seedAndBlockNumber is zero then the response has been fulfilled
-			// and we should skip it
-			rlog.Infow("Request already fulfilled", "callback", callback)
-			lsn.markLogAsConsumed(req.lb)
-			processed[vrfRequest.RequestId.String()] = struct{}{}
-			continue
+		chunk := unconsumed[chunkStart:chunkEnd]
+
+		var unfulfilled []pendingRequest
+		alreadyFulfilled := lsn.checkReqsFulfilled(ctx, l, chunk)
```

Comment:
Done. Returning err from the already-fulfilled check, and short circuiting if it's canceled. 

For the pipeline errs still relying on the individual pipeline errs which would be "cancelled" in this case, then it will short circuit on the next iteration. Could also add a ctx.Err() check on each loop, not sure what best practice is here.